### PR TITLE
SELC-936 eventhub

### PR DESCRIPTION
**Pull request by antonioT90**
### List of changes

Added eventhub connection to publish events towards datalake external system

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [x] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

**Review by andrea-putzu — DISMISSED** [review]

File: src/core/env/prod/terraform.tfvars [hunk 1]
```diff
@@ -108,4 +109,20 @@ dns_ns_interop_selfcare = [
...
\ No newline at end of file
+]
+
+## Event hub
+eventhub_sku_name                 = "Standard"
+eventhub_capacity                 = 2
+eventhub_auto_inflate_enabled     = true
+eventhub_maximum_throughput_units = 4
+eventhub_zone_redundant           = true
+eventhub_alerts_enabled           = false
```
Comment (diegolagospagopa) [inline src/core/env/prod/terraform.tfvars:120]: can be `true`?

File: src/core/events.tf [hunk 2]
```diff
@@ -0,0 +1,85 @@
+resource "azurerm_resource_group" "event_rg" {
+  name     = format("%s-event-rg", local.project)
```
Comment (diegolagospagopa) [inline src/core/events.tf:2]: Please use the interpolation format for string

File: src/core/events.tf [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  location = var.location
+
+  tags = var.tags
+}
+
+module "eventhub_snet" {
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v2.18.6"
+  name                                           = format("%s-eventhub-snet", local.project)
```
Comment (diegolagospagopa) [inline src/core/events.tf:10]: please use interpolation for strings

File: src/core/events.tf [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  address_prefixes                               = var.cidr_subnet_eventhub
+  resource_group_name                            = azurerm_resource_group.rg_vnet.name
+  virtual_network_name                           = module.vnet.name
+  service_endpoints                              = ["Microsoft.EventHub"]
+  enforce_private_link_endpoint_network_policies = true
+}
+
+module "event_hub" {
+  source                   = "git::https://github.com/pagopa/azurerm.git//eventhub?ref=v2.18.7"
+  name                     = format("%s-eventhub-ns", local.project)
```
Comment (diegolagospagopa) [inline src/core/events.tf:20]: please use interpolation for strings

File: src/core/variables.tf [hunk 5]
```diff
@@ -894,3 +899,118 @@ variable "robots_indexed_paths" {
...
+        },
+        {
+          name   = "datalake"
+          listen = true
+          send   = false
+          manage = false
+        }
+      ]
+    }
+  ]
```
Comment (diegolagospagopa) [inline src/core/variables.tf:978]: Is possible to put it into the variables files, and not into defaults to be more simple to debug in the future?

File: src/k8s/selc_secrets.tf [hunk 6]
```diff
@@ -214,3 +214,21 @@ resource "kubernetes_secret" "common-secrets" {
...
 }
+
+resource "kubernetes_secret" "event-secrets" {
+  metadata {
+    name      = "event-secrets"
+    namespace = kubernetes_namespace.selc.metadata[0].name
+  }
+
+  data = {
+    KAFKA_BROKER            = "${format("%s-eventhub-ns", local.project)}.servicebus.windows.net:9093"
```
Comment (diegolagospagopa) [inline src/k8s/selc_secrets.tf:225]: Please use the interpolation format for string

File: src/core/env/prod/terraform.tfvars (hunk 1 above)
Comment (antonioT90) [reply src/core/env/prod/terraform.tfvars:120]: We are just publishing into this queue, the datalake will read from it, but asked us to host it.
We have nothing to monitor here.

Comment (diegolagospagopa) [reply src/core/env/prod/terraform.tfvars:120]: OK

**Review by diegolagospagopa — APPROVED** [review]

**Pull request merged**